src/api/transaction/settings.js

```
@@ -5,32 +5,23 @@ const _ = require('lodash');
 const utils = require('./utils');
 const ripple = utils.common.core;
 const validate = utils.common.validate;
-const constants = utils.common.constants;
-const InvalidRequestError = utils.common.errors.InvalidRequestError;
+const AccountFlagIndices = utils.common.constants.AccountFlagIndices;
+const AccountFields = utils.common.constants.AccountFields;
+const ValidationError = utils.common.errors.ValidationError;
 
 // Emptry string passed to setting will clear it
 const CLEAR_SETTING = '';
 
-/**
- * Set integer flags on a transaction based on input and a flag map
- *
- * @param {Transaction} transaction
- * @param {Object} input - Object whose properties determine whether
- *                to update the transaction's SetFlag or ClearFlag property
- * @param {Object} flags - Object that maps property names to transaction
- *                integer flag values
- *
- * @returns undefined
- */
-function setTransactionIntFlags(transaction, values, flags) {
-  for (let flagName in flags) {
+function setTransactionFlags(transaction, values) {
+  const nameToIndexMap = AccountFlagIndices;
+  for (let flagName in nameToIndexMap) {
```

SetFlag and ClearFlag can only be set once for each transaction? I haven't looked elsewhere at calling code, but is it possible that some flags could be silently ignored?